### PR TITLE
Add a script to build a base system image based on Clear Linux

### PR DESCRIPTION
**Pull request by andrisaar**
Last piece of the puzzle for now: a script to build a tarball suitable for use as the system image in Oak Docker.

### CODE REVIEW

File: oak_containers_system_image/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM clearlinux:base
+
+# Remove .dockerenv as we will not run this image in an actual Docker container.
+RUN rm /.dockerenv
+
+# Don't bother starting the graphical interface, let's stick with the basic multi-user target.
+RUN rm /usr/lib/systemd/system/default.target
+RUN ln -s /usr/lib/systemd/system/multi-user.target /usr/lib/systemd/system/default.target
```
Comment (conradgrobler) [inline oak_containers_system_image/Dockerfile:8]: nit: line break missing at the end of the file

File: oak_containers_system_image/build.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+docker build . --tag oak-containers-system-image
+# We need to actually create a container, otherwise we won't be able to use `docker export` that gives us a filesystem image.
+# (`docker export` creates a tarball which has all the layers separate, which is _not_ what we want.)
+docker run oak-containers-system-image:latest
+readonly NEW_DOCKER_CONTAINER_ID="$(docker ps -a -f 'ancestor=oak-containers-system-image:latest' --format '{{.ID}}')"
+
+mkdir -p target
+docker export $NEW_DOCKER_CONTAINER_ID > target/image.tar
```
Comment (conradgrobler) [inline oak_containers_system_image/build.sh:10]: nit: the variable should be double-quoted

File: oak_containers_system_image/build.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+docker build . --tag oak-containers-system-image
+# We need to actually create a container, otherwise we won't be able to use `docker export` that gives us a filesystem image.
+# (`docker export` creates a tarball which has all the layers separate, which is _not_ what we want.)
```
Comment (conradgrobler) [inline oak_containers_system_image/build.sh:5]: nit: should this be `docker build`?

File: oak_containers_system_image/build.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+docker run oak-containers-system-image:latest
```
Comment (conradgrobler) [inline oak_containers_system_image/build.sh:6]: do you need to use `docker run` here? Is it not possible to use docker create as in https://github.com/project-oak/oak/blob/main/oak_containers_container_example/build_container_bundle?

**Review by conradgrobler — APPROVED** [review]

File: oak_containers_system_image/build.sh (hunk 3 above)
Comment (andrisaar) [reply oak_containers_system_image/build.sh:5]: It should be `docker save`.

File: oak_containers_system_image/build.sh (hunk 4 above)
Comment (andrisaar) [reply oak_containers_system_image/build.sh:6]: If you look at Juliette's script, there's a `docker create` hidden in there.

However, indeed, using `docker create` is better than `docker run` as we don't really need to _run_ the container from the image we've built. Thanks for the link!

**Pull request merged**